src/main/scala/com/databricks/labs/overwatch/utils/PipelineValidationHelper.scala

```diff
@@ -0,0 +1,498 @@
+
+    val validation = validateNullRuleSet.validate()
+
+    val completeReportDF = validation.completeReport
+
+    validateNullRuleSet.getRules.foreach(elem => {
+      val colName = elem.inputColumn.toString.split("\\(")(1).split("\\)")(0).split(" ")(0)
+      val healthCheckRuleColumn = elem.ruleName
+      val dfWithNegativeValidation = completeReportDF.filter((col(s"$healthCheckRuleColumn.passed") === false) ||
+        col(s"$healthCheckRuleColumn.passed").isNull).select(keys.map(col): _*)
+      val countOfNegativeValidation = dfWithNegativeValidation.count()
+      if (countOfNegativeValidation == 0) {
+        val healthCheckMsg = "Success"
+        vStatus.append(HealthCheckReport(etlDB, table_name, healthCheckRuleColumn,"Single_Table_Validation", Some(healthCheckMsg), Overwatch_RunID))
+      } else {
+        val (healthCheckMsg: String, healthCheckType: String) =
+          if (validationType.toLowerCase() == "validate_greater_than_zero") {
+            (s"HealthCheck Failed: got $countOfNegativeValidation ${colName}s which are not greater than zero or is NULL", "Failure")
+          } else if (validationType.toLowerCase() == "validate_not_null") {
+            (s"HealthCheck Failed: got $countOfNegativeValidation ${colName}s which are null", "Failure")
+          } else if (validationType.toLowerCase() == "validate_leq_one") {
+            (s"HealthCheck Failed: got $countOfNegativeValidation ${colName}s which are greater than 1", "Failure")
+          } else if (validationType.toLowerCase() == "validate_values_in_between") {
+            (s"HealthCheck Warning: got $countOfNegativeValidation ${colName}s which are not in between expected values", "Warning")
+          }
+          else {
+            (s"HealthCheck Warning : got $countOfNegativeValidation ${colName}s which are greater than 1", "Warning")
+          }
+        vStatus.append(HealthCheckReport(etlDB, table_name, healthCheckRuleColumn,"Single_Table_Validation",Some(healthCheckMsg), Overwatch_RunID))
+        dfWithNegativeValidation.toJSON.collect().foreach(jsonString => {
+          qStatus.append(QuarantineReport(etlDB, table_name, healthCheckRuleColumn,"Single_Table_Validation", healthCheckType, jsonString))
+        })
+      }
+    })
+    validationStatus ++= vStatus
+    quarantineStatus ++= qStatus
+
+    (validationStatus, quarantineStatus)
+  }
+
+  def validateColumnBetweenMultipleTable(
+                                          source: PipelineTable,
+                                          target: PipelineTable,
+                                          sourceDF: DataFrame,
+                                          targetDF: DataFrame,
+                                          column: String,
+                                          key: Array[String],
+                                          validationStatus: ArrayBuffer[HealthCheckReport],
+                                          quarantineStatus: ArrayBuffer[QuarantineReport]
```

Comment:
it will be more helpful if you can write a little description for the functions

Reply:
Done